api/config/db.ts

```diff
@@ -1,12 +1,10 @@
-require("dotenv").config();
-const mongoose = require("mongoose");
+import dotenv from "dotenv";
+dotenv.config();
+import mongoose from "mongoose";
 
 export const connectToDatabase = async () => {
   try {
-    return await mongoose.connect(process.env.MONGO_URI, {
-      useNewUrlParser: true,
-      useUnifiedTopology: true,
-    });
+    return await mongoose.connect(process.env.MONGO_URI!);
```

Comment:
Be careful with using `!`
Here, the correct solution is going to be what @nahuelRo is working on with Zod.